react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+import { ImageCard } from './ImageCard';
+import { NextButton } from './NextButton';
+import { Dots } from './Dots';
+import { PrevButton } from './PrevButton';
+
+type Props = {
+  images: { src: string; alt: string }[];
+};
+
+export function Carousel({ images }: Props) {
+  const [activePicture, setActivePicture] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setActivePicture((prevIndex) => (prevIndex + 1) % images.length);
+    }, 3000);
+    return () => clearInterval(interval);
+  }, [images.length]);
```

Comment:
Rather than making `images.length` the dependency, we'd just do `images`. We don't necessarily care about if the length changes, but we do care if the images passed in change